mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.linear_model import Ridge
+from .base import get_coefs
+
+
+class ReceptiveField(object):
+    """Fit a receptive field model.
+
+    Parameters
+    ----------
+    stim_features : array, shape (n_features,)
+        Names for the input features to the model.
+    lags : array, shape (n_lags)
+        Time lags to be included in the model. This will be combined
+        with `sfreq` to convert these values into indices in the data.
+        If `sfreq` == 1, then these should be indices. If not, then
+        time lags should be in seconds.
+    sfreq : float
+        The sampling frequency of the data.
+    model : instance of sklearn model | None
+        The model used in fitting inputs and outputs. This can be any
+        sklearn-style model that contains a fit and predict method.
+    """
+    def __init__(self, stim_features, lags, sfreq=1., model=None):
+        self.stim_features = stim_features
+        self.lags = lags
+        self.model = Ridge() if model is None else model
+        self.sfreq = sfreq
+
+    def fit(self, X, y):
+        """Fit a receptive field model.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
```

Comment:
Since you're not using this in the example/tutorials, and in light of @agramfort 's point on samples being the first dimension, I would remove the support for epochs for now, and only works with X shape = n_times, n_channels.